File: Vibechat.Web/Vibechat.Shared/DTO/Messages/ChatEvent.cs
using Newtonsoft.Json;

namespace Vibechat.Shared.DTO.Messages
{
    [JsonObject(MemberSerialization.OptOut)]
    public class ChatEvent
    {
        public string Actor { get; set; }
        
        public ChatEventType Type { get; set; }

        public string UserInvolved { get; set; }

        public string ActorName { get; set; }

        public string UserInvolvedName { get; set; }
    }
}

File: Vibechat.Web/Vibechat.Web/ClientApp/src/app/Data/ChatEvent.ts
import {AppUser} from "./AppUser";
import {ChatEventType} from "./ChatEventType";

export class ChatEvent{

  public constructor(init?: Partial<ChatEvent>) {
    (<any>Object).assign(this, init);
  }

  //Why or 'because of who' this event occured
  public actor: string;
  //banned, kicked, joined, invited, left
  public type: ChatEventType;

  public userInvolved: string;

  public actorName: string;

  public userInvolvedName: string;
}
